### PR TITLE
[BQ] Updated queries

### PR DESCRIPTION
**Pull request by chejennifer**
- hide bq and download buttons when no timeline chart
- also hide bq button when all timeline charts have the "delta" option selected
- update timeline queries to allow for combinations of whether or not per capita is selected and whether or not source is selected
- update map queries to allow for combinations of whether or not per capita is selected, whether or not source is selected, and whether or not date is selected.
- revert url change from [pr #1570](https://github.com/datacommonsorg/website/pull/1570)

### CODE REVIEW

**Review by pradh — APPROVED** [review]
Phew!

File: static/js/tools/shared/bq_utis.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { StatMetadata } from "../../shared/stat_types";
```
Comment (beets) [inline static/js/tools/shared/bq_utis.ts:1]: please fix (should be after license)

File: static/js/tools/timeline/chart_region.tsx [hunk 2]
```diff
@@ -50,10 +55,12 @@ class ChartRegion extends Component<ChartRegionPropsType> {
   bulkDownloadLink: HTMLAnchorElement;
   bqLink: HTMLAnchorElement;
   allStatData: { [key: string]: StatData };
+  metadataMap: Record<string, Record<string, StatMetadata>>;
```
Comment (beets) [inline static/js/tools/timeline/chart_region.tsx:58]: nit: mind adding a comment describing the keys and values of the map?

File: static/js/tools/timeline/chart_region.tsx [hunk 3]
```diff
@@ -233,22 +258,46 @@ class ChartRegion extends Component<ChartRegionPropsType> {
...
     }
-    const placeSvQueryString = placeSvPairs
-      .map(
-        (placeSvPair) =>
-          `(O.observation_about = "${placeSvPair[0]}" AND O.variable_measured = "${placeSvPair[1]}")`
-      )
+    return false;
+  }
+
+  private getQueryForGroup(
```
Comment (beets) [inline static/js/tools/timeline/chart_region.tsx:270]: please add a todo here for tests (and there might be opportunity for slight refactoring).

**Review by beets — APPROVED** [review]
nothing blocking! changes lgtm

File: static/js/tools/shared/bq_utis.ts [hunk 4]
```diff
@@ -36,3 +38,29 @@ export function setUpBqButton(getSqlQuery: () => string): HTMLAnchorElement {
...
+    ? `= '${metadata.measurementMethod}'`
+    : "IS NULL";
+  const unitString = metadata.unit ? `= '${metadata.unit}'` : "IS NULL";
+  const obsPeriodString = metadata.observationPeriod
+    ? `= '${metadata.observationPeriod}'`
+    : "IS NULL";
+  const sFactorString = metadata.scalingFactor
+    ? `= '${metadata.scalingFactor}'`
+    : "IS NULL";
+  return `${obsTableName}.measurement_method ${mMethodString} AND
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply static/js/tools/shared/bq_utis.ts:62]: yes, we should include import name but I was looking at the Observation table and it looks like import name isn't a column?

[the remark this answers is not in the document]
Comment (pradh) [reply static/js/tools/shared/bq_utis.ts:63]: It needs to be joined with the Provenance table, based on the prov_id column.

We can add that in the next pass...

File: static/js/tools/map/bq_query_utils.ts [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+): string {
+  return (
+    `WITH ${LATEST_OBS_DATE_VIEW_NAME} AS (` +
+    `
+  ${getChildPlaceView(place)}
+  SELECT O.variable_measured as StatVarId,
+        O.observation_about as PlaceId,
+        MAX(O.Observation_date) as LatestDate
+  FROM \`data_commons.Observation\` as O
+  JOIN ChildPlace ON TRUE
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply static/js/tools/map/bq_query_utils.ts]: done

File: static/js/tools/timeline/chart_region.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/tools/timeline/chart_region.tsx:59]: done

File: static/js/tools/timeline/chart_region.tsx (hunk 3 above)
Comment (chejennifer) [reply static/js/tools/timeline/chart_region.tsx:271]: I had already put a todo for getSqlQuery() which uses this function

**Pull request merged**